pkg/machine/apple/apple.go

```diff
@@ -275,6 +277,28 @@ func StartGenericAppleVM(mc *vmconfigs.MachineConfig, cmdBinary string, bootload
 
 	logrus.Debugf("helper command-line: %v", cmd.Args)
 
+	if mc.LibKrunHypervisor != nil && logrus.IsLevelEnabled(logrus.DebugLevel) {
+		rtDir, err := mc.RuntimeDir()
+		if err != nil {
+			return nil, nil, err
+		}
+		kdFile, err := rtDir.AppendToNewVMFile("krunkit-debug.sh", nil)
+		if err != nil {
+			return nil, nil, err
+		}
+		f, err := os.Create(kdFile.Path)
+		if err != nil {
+			return nil, nil, err
+		}
+		os.Chmod(kdFile.Path, 0744)
+
+		f.WriteString("#!/bin/sh\n")
+		f.WriteString(fmt.Sprintf("exec %s", strings.Join(cmd.Args, " ")))
```

Comment:
strings.Join() on the Args will not work, this does not handle special chars/spaces etc... and will just break with hard to diagnose errors.

I am not sure what the purpose of spawning another terminal is, couldn't we just attach it to the current podman machine start command stdout/err?

Reply:
> strings.Join() on the Args will not work, this does not handle special chars/spaces etc... and will just break with hard to diagnose errors.

What's the preferred way to do it?

> I am not sure what the purpose of spawning another terminal is, couldn't we just attach it to the current podman machine start command stdout/err?

I think the UX is significantly better this way. Stealing podman's stdout/stdin feels weird, I would expect podman to display debugging messages and not be interactive, just as happens with other provides.

Reply:
> > strings.Join() on the Args will not work, this does not handle special chars/spaces etc... and will just break with hard to diagnose errors.
> 
> What's the preferred way to do it?

I don't think there is any ready to use interface for writing string slices arguments as shell escaped args. Looping over the args and using fmt.Sprintf("%q",arg) may work although I am unsure if it handles quoting like the shell.

But maybe stepping back why do you have to create this tmp script anyway, can't you append the args to the terminal command?

Reply:
> > I am not sure what the purpose of spawning another terminal is, couldn't we just attach it to the current podman machine start command stdout/err?
> 
> I think the UX is significantly better this way. Stealing podman's stdout/stdin feels weird, I would expect podman to display debugging messages and not be interactive, just as happens with other provides.

But isn't this a fundamental different behaviour here, you are spawning the full krun in another terminal unlike vfkit which is started as direct child and then opens a GUI window.

We don't do it today but I think we should read stderr in order to return the error strings to the user if something fails. I don't see how this would work with your solution.

Reply:
> I don't think there is any ready to use interface for writing string slices arguments as shell escaped args. Looping over the args and using fmt.Sprintf("%q",arg) may work although I am unsure if it handles quoting like the shell.

Did it this way in v2, thanks!

> But maybe stepping back why do you have to create this tmp script anyway, can't you append the args to the terminal command?

On macOS, `/usr/bin/open` allows you to pass arguments to Terminal itself but, AFAIK, there isn't a way to tell Terminal to pass any arguments to command being executed on it. Ref: https://stackoverflow.com/questions/29510815/how-to-pass-command-line-arguments-to-a-program-run-with-the-open-command